client/app/bundles/course/learning-map/containers/Canvas/index.jsx

```diff
@@ -0,0 +1,104 @@
+import { useRef, useState } from 'react';
+import { TransformWrapper, TransformComponent } from 'react-zoom-pan-pinch';
+import { Xwrapper } from 'react-xarrows';
+import Levels from '../Levels';
+import ArrowOverlay from '../ArrowOverlay';
+import ZoomActionElements from '../../components/ZoomActionElements';
+
+const styles = {
+  cursorPosition: {
+    position: 'absolute',
+  },
+  transformComponentWrapper: {
+    width: '100%',
+  },
+  wrapper: {
+    outline: 'none',
+    overflow: 'hidden',
+    position: 'relative',
+    width: '100%',
+    zIndex: 1,
+  },
+  zoomAnimation: {
+    animationTime: 300,
+    animationType: 'easeOutCubic',
+  },
+};
+
+const Canvas = () => {
+  const gateInputSizeThreshold = 5;
```

Comment:
Perhaps can shift all these constants out of this component. Like right above it.

Reply:
Done.